### ISSUE TITLE
Waterline 0.10.11 passing array as string to afterUpdate

### ISSUE THREAD

**Issue by chadxz** [balderdashy/waterline#698]
When I update a record in the database (using 'sails-mysql' adapter), the `values` passed to the afterUpdate lifecycle callback looks like this:

```
{ id: '4d094885-7e6f-4454-91f0-6a55e3d1b6eb',
  accountId: 'webhooksControllerTestAccountId',
  appId: 'webhooksControllerTestAppId',
  endpointId: 'updatedEndpointId',
  webhookURL: 'updatedWebhookURL',
  groups: '["updatedGroups"]',
  createdAt: Thu Oct 23 2014 15:34:12 GMT-0500 (CDT),
  updatedAt: Thu Oct 23 2014 15:34:12 GMT-0500 (CDT) }
```

As you can tell from the model definition below, the groups field is an array. I have checked to ensure that i'm passing it an array when its going in, and reading it back out of the database the value is an array, but the value passed to the afterUpdate is a string

I can probably fix this and issue a PR, but i was wondering where I should start looking for the issue... would the code responsible for serializing arrays to string and back be in waterline or sails-mysql?

model definition:
```
module.exports = {

    attributes: {
        id: {
            type: 'string',
            primaryKey: true
        },
        accountId: {
            type: 'string',
            required: true
        },
        appId: {
            type: 'string',
            required: true
        },
        endpointId: {
            type: 'string',
            required: true
        },
        webhookURL: {
            type: 'string',
            required: true
        },
        groups: {
            type: 'array',
            defaultsTo: []
        },
        toJSON: function () {
            var obj = this.toObject();
            delete obj.createdAt;
            delete obj.updatedAt;
            return obj;
        }
    }
```

**Comment by devinivy** [comment]
As I understand it, JSON/Array data types should be passed to the afterUpdated lifecycle as an object/array respectively, rather than a string.  That sounds reasonable.  @chadxz mind changing the title to reflect this?  Anybody mind writing a failing test [here](https://github.com/balderdashy/waterline/tree/master/test/unit/callbacks)?

**Comment by chadxz** [comment]
@devinivy on it

**Comment by chadxz** [comment]
@devinivy this doesn't seem to be an issue with the waterline layer, but of the sails-mysql adapter. Waterline is just passing the values to afterUpdate what it got from the adapter.

**Comment by chadxz** [comment]
i'm gonna keep trying for a bit

**Comment by chadxz** [comment]
nevermind, i'm seeing the issue now

**Comment by chadxz** [comment]
@devinivy https://github.com/chadxz/waterline/commit/88407fc987b62f196d92fdc580df59e8d6ff8c23 how's this?

**Comment by devinivy** [comment]
Looks good!  I made one comment on it.  Thanks, @chadxz!!

**Comment by devinivy** [comment]
Here's where the code change should likely happen: https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/dql/update.js#L207-L226

**Comment by chadxz** [comment]
@devinivy okay, i fixed that and re-wrote the commit. new commit is https://github.com/chadxz/waterline/commit/a90fa94561a60b63a0ef89776587d407e7df3ec0

**Comment by dmarcelino** [comment]
Yes @devinivy, I think that make sense. @chadxz, do you have something you can PR? Would like to see that.

I wonder if this is in anyway related to #477.

**Comment by chadxz** [comment]
I don't have a fix, only the failing test illustrating the problem. Do you want me to PR that?

**Comment by dmarcelino** [comment]
Yes please, it won't be merged until we have a fix but at least we'll be able to use and discuss the test which is a great help! Thanks.

**Comment by dmarcelino** [comment]
Thanks for #971 @chadxz.

**Comment by chadxz** [comment]
np!

**Comment by dmarcelino** [comment]
@chadxz, @devinivy, I've looked a bit deeper into this and I believe this is a sails-mysql issue, sails-mysql must be returning the wrong values after an update.

I've created a new test in balderdashy/waterline-adapter-tests#73 which reproduces this problem in sails-mysql while passing on all other adapters. I also raised balderdashy/sails-mysql#219 to deal this.

If you guys agree lets close this and continue the discussion in balderdashy/sails-mysql#219.
